bip-???-cat.mediawiki

```diff
@@ -0,0 +1,91 @@
+<pre>
+  BIP: ???
+  Layer: Consensus (soft fork)
+  Title: OP_CAT
+  Author: Ethan Heilman <ethan.r.heilman@gmail.com>
+          Armin Sabouri <arminsdev@gmail.com>
+  Status: Draft
+  Type: Standards Track
+  Created: 2023-10-21
+  Comments-URI: https://github.com/bitcoin/bips/wiki/Comments:BIP-op-cat
+  License: BSD-3-Clause
+</pre>
+
+==Abstract==
+
+This BIP reintroduces OP_CAT in the form of a new tapscript opcode which allows the concatenation of two values on the stack. This opcode would be activated via a soft fork by redefining the opcode OP_SUCCESS126.
+
+When evaluated the OP_CAT instruction:
+# Pops the top two values off the stack,
+# concatenates the popped values together,
+# and then pushes the concatenated value on the top of the stack.
+
+OP_CAT fails if there are less than two values on the stack or if a concatenated value would have a combined size greater than the maximum script element size of 520 Bytes.
```

Comment:
```suggestion
OP_CAT fails if there are less than two values on the stack or if a concatenated value would have a combined size greater than the maximum script element size of 520 bytes.
```

Reply:
Original fine. Tiny preference for the uncapitalized 'bytes' too, though.

Reply:
I've yet to find a source claiming the capitalized version is correct. And in particular, the BIP repository uses exclusively the lowercase version with [two exceptions in BIP136](https://github.com/bitcoin/bips/blob/master/bip-0136.mediawiki#tx-position-value).